### PR TITLE
prov/rxm: Register bounce buffers with HMEM

### PR DESCRIPTION
**Pull request by iziemba**
Registering all HMEM bounce buffers with HMEM can lead to decreased
memcpy times.

Signed-off-by: Ian Ziemba <ian.ziemba@hpe.com>

### CODE REVIEW

File: prov/rxm/src/rxm_ep.c [hunk 1]
```diff
@@ -113,6 +113,11 @@ static int rxm_buf_reg(struct ofi_bufpool_region *region)
 	struct rxm_domain *rxm_domain;
 	int ret;
 
+	ret = ofi_hmem_host_register(region->mem_region,
```
Comment (shefty) [inline prov/rxm/src/rxm_ep.c:116]: I think we need to check if FI_HMEM has been enabled by the application prior to calling this.

Comment (iziemba) [reply prov/rxm/src/rxm_ep.c:116]: Can do.

**Comment by iziemba** [comment]
@shefty Looks like Intel CI failed.

**Comment by shefty** [comment]
Intel CI failure:
```
06:44:13  - name:   fi_unexpected_msg -e rdm -i 10 -p "shm"
06:44:13    result: Fail
06:44:13    time:   301
06:44:13    server_cmd: /home/build/ofi-Install/libfabric/ofi_libfabric/PR-6248/2/dbg/bin/fi_unexpected_msg -e rdm -i 10 -p "shm"  -s n7
06:44:13    server_stdout: |
06:44:13      Killed by signal 15.
06:44:13    client_cmd: /home/build/ofi-Install/libfabric/ofi_libfabric/PR-6248/2/dbg/bin/fi_unexpected_msg -e rdm -i 10 -p "shm"  -s n8 n7
06:44:13    client_stdout: |
06:44:13      Killed by signal 15.
```

**Comment by shefty** [comment]
@aingerson - The CI failure is unrelated to this PR, but looks like a real failure to investigate.  It may be difficult to reproduce, since this is the first time I've seen it.

**Review by shefty — APPROVED** [review]

**Pull request merged**